docs/configuration.md

```diff
@@ -3134,3 +3134,119 @@ The stage level scheduling feature allows users to specify task and executor res
+
+# Push-based shuffle overview
+
+Push-based shuffle is an improved shuffle architecture that optimizes the reliability and performance of the shuffle step in Spark. Complementing the existing shuffle mechanism, push-based shuffle takes a best-effort approach to push the shuffle blocks generated by the map tasks to remote shuffle services to be merged per shuffle partition. When the reduce tasks start running, they fetch a combination of the merged shuffle partitions and some of the original shuffle blocks to get their input data. As a result, push-based shuffle converts shuffle services’ small random disk reads into large sequential reads. The reduce tasks are also scheduled with locality preferences of the locations of their corresponding merged shuffle partitions, which helps to significantly improve shuffle fetch data locality.
+
+<p> <b> Currently push-based shuffle is only supported for Spark on YARN. </b></p>
+
+### Shuffle server side configuration options
+
+<table class="table">
+<tr><th>Property Name</th><th>Default</th><th>Meaning</th><th>Since Version</th></tr>
+<tr>
+  <td><code>spark.shuffle.server.mergedShuffleFileManagerImpl</code></td>
+  <td>
+    <code>org.apache.spark.network.shuffle.ExternalBlockHandler$NoOpMergedShuffleFileManager</code>
+  </td>
+  <td>
+    Class name of the implementation of MergedShuffleFileManager that manages push-based shuffle. This acts as a server side config to disable or enable push-based shuffle. By default, push-based shuffle is disabled at the server side, which does not perform push-based shuffle irrespective of the client side configuration. To turn on push-based shuffle at the server side, set the configuration to <code>org.apache.spark.network.shuffle.RemoteBlockPushResolver</code>
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.server.minChunkSizeInMergedShuffleFile</code></td>
+  <td><code>2m</code></td>
+  <td>
+    The minimum size of a chunk when dividing a merged shuffle file into multiple chunks during push-based shuffle. A merged shuffle file consists of multiple small shuffle blocks. Fetching the complete merged shuffle file in a single disk I/O increases the memory requirements for both the clients and the shuffle service. Instead, the shuffle service serves the merged file in <code> MB-sized chunks </code>. This configuration controls how big a chunk can get. A corresponding index file for each merged shuffle file will be generated indicating chunk boundaries.
```

Comment:
`...clients and the shuffle service.` -> `...clients and the external shuffle service.`